app/src/main/java/de/westnordost/streetcomplete/map/MainFragment.kt

```diff
@@ -233,7 +233,10 @@ class MainFragment : Fragment(R.layout.fragment_main),
     /* ---------------------------------- MapFragment.Listener ---------------------------------- */
 
     override fun onMapInitialized() {
-        gpsTrackingButton.isActivated = mapFragment?.isFollowingPosition ?: false
+        val isFollowingPosition = mapFragment?.isFollowingPosition ?: false
+        val isPositionKnown = mapFragment?.displayedLocation != null ?: false
+        gpsTrackingButton.isActivated = isFollowingPosition
+        gpsTrackingButton.visibility = if (isFollowingPosition && isPositionKnown) View.INVISIBLE else View.VISIBLE
```

Comment:
I thought so too and initially did it like this, however I think this would not work for the code in `onLocationChanged` because that method is called once the Android system reports that it found a new location. The `LocationAwareMapFragment` is also listening on location updates to set its `displayedLocation`  and it is not guaranteed that the `LocationAwareMapFragment` will get the call before the `MainFragment` does. So, it is possible that `mapFragment.displayedLocation` is still `null` when `onLocationChanged` is called on the `MainFragment`.